### DISCUSSION TITLE
Add support for Tasmesh in Home Assistant Discovery SO19

### DISCUSSION THREAD

**Discussion by jenniferlee1818** [arendst/Tasmota#15221] [Ideas]
I recently started using the TasMesh which is awesome because it takes many of my Tasmota devices off of my router.  But with the LWT=Offline they don't work in Home Assistant.  My workaround is to use the Tasmota integration plus add all 125 entities manually (platform: mqtt) in configuration.yaml.  Whenever the LWT=Offline, I publish LWT2=Online to my mqtt entities (and vice versa) where LWT2 is the availability_topic.  Hack  _**assuming**_ LWT=Offline means connected to the TasMesh.  Now in Home Assistant, if the Tasmota device is connected to my router, its entities are available in the Tasmota integration, and whenever the Tasmota device is connected to the mesh, its entities are available in the MQTT integration.  Then, I use conditional cards in the dashboard, and use both entities as automation triggers, actions, etc.  This all seems so messy and it's so much extra work.  What is the easiest way to make Tasmesh nodes work with Home Assistant?  Can we get a SetOption19 2? Or can the Tasmota integration be made to include mesh nodes?  I don't find much about TasMesh online.  Am I the only one doing this?

**Comment by barbudor** [comment]
Hi
Never used TasMesh, and I don't think many people are. So be summoning @sillyfrog is better.

However, from what I read in the docs, TasMesh acts either as a router across networks or a NAT router.
So I would imagine that an ESP client connected to the TasMesh router would be able to publish any topic to the MQTT broker just like if it was directly connected to the same network, including LWT=Online.

Which mode are you using ? router or NAT ?

**Comment by sillyfrog** [comment]
If you are using the current version of Home Assistant, I would suggest you turn _off_ SetOption19, and use the [native Tasmota Integration ](https://tasmota.github.io/docs/Home-Assistant/)(with that many devices, the migration will... suck... I have about 30 and it was painful). But once it's done, things "just work" in HA, all the devices appear and are automatically discovered using Tasmota MQTT topics. I _think_ that'll work, but I've not actually used Tasmota Mesh for "production".

That means that if you are using NAT or a Router, as the topics are published to MQTT, HA does the right thing. I'm obviously using https://tasmota.github.io/docs/Range-Extender/ , but I have my network configured now for Router mode to give me the option of direct web interface access, but it's a pain to setup initially however. Setting up for NAT makes things much easier. Unlike Tasmesh, you need to design and implement the network topology yourself (ie: which nodes will be "routers" and which will connect to those). An advantage over Tasmesh is it works on the ESP8266 (acting as both a router or NAT), and you can connect any device to the new SSID.

(Hopefully that helps/guides you in the right direction, sorry if I missed the mark.)

**Reply by jenniferlee1818** [reply]
> If you are using the current version of Home Assistant, I would suggest you turn _off_ SetOption19, and use the [native Tasmota Integration ](https://tasmota.github.io/docs/Home-Assistant/)(with that many devices, the migration will... suck... I have about 30 and it was painful). But once it's done, things "just work" in HA, all the devices appear and are automatically discovered using Tasmota MQTT topics. I _think_ that'll work, but I've not actually used Tasmota Mesh for "production".

SO19 0 doesn't work with TasMesh.  Whenever a device connects to the TasMesh broker, it becomes unavailable in the Tasmota integration.

[https://github.com/arendst/Tasmota/blob/development/info/xdrv_57_tasmesh.md](https://github.com/arendst/Tasmota/blob/development/info/xdrv_57_tasmesh.md) > Limitations: As there is no direct connection from the node to the MQTT broker it will signal the node as LWT Offline

**Reply by sfromis** [reply]
With current Tasmota versions, using `SetOption19` is no longer relevant, unless you have done a custom build to include the deprecated older Tasmota support for HA MQTT Discovery instead of the current Tasmota integration, which is now included in builds.

**Reply by ferbulous** [reply]
With so19 not relevant, is there an alternative to keep the nodes status as online in homeassistant once connected to the esp32 broker?

**Comment by ferbulous** [comment]
Hi, do you mind sharing some example for the yaml with LWT2 hack and the conditional cards?
Facing the same problem here with the devices going offline
